doc/rtd/reference/cli.rst

```diff
@@ -487,4 +498,52 @@ Which would produce the following example output:
       "status": "done"
     }
 
+cloud-id
+========
+
+``cloud-id`` reports the canonical cloud-id for the instance.
+
+For example, on LXD, we'll see:
+
+.. code-block:: shell-session
+
+   $ cloud-id
+   lxd
+
+Whereas on AWS, we'll see:
+
+.. code-block:: shell-session
+
+   $ cloud-id
+   aws
+
+See the
+`cloud-id man page <https://manpages.ubuntu.com/manpages/noble/en/man1/cloud-id.1.html>`_
+for more details.
+
+.. _cloud-init-per:
+
+cloud-init-per
+==============
+
+``cloud-init-per`` will run a command with arguments at a specific frequency.
+
+For example, with the following command:
+
+.. code-block:: shell-session
+
+   $ cloud-init-per once hello bash -c 'echo "Hello, world!" >> /tmp/hello'
```

Comment:
This command isn't very useful outside of boot scripts, right? Do you think that we should add the intended purpose?

One question from the man page:

> instance: run only the first boot for a given instance-id

This makes me wonder what would happen if an instance has already booted and I ran this command. Maybe that doesn't matter?

Reply:
> This command isn't very useful outside of boot scripts, right? Do you think that we should add the intended purpose?

Or something that runs generally every boot, but yeah, I guess that would usually be boot scripts. 

> This makes me wonder what would happen if an instance has already booted and I ran this command. 

It'll do the thing, write a semaphore file, and then do the thing on subsequent boots. I can update the man page to be more precise.

Reply:
I updated the man page, but to more directly answer

> This command isn't very useful outside of boot scripts, right? Do you think that we should add the intended purpose?

I personally find it harder to grok when embedded in more complicated scripts, which is why I went with a basic hello world example. I think that's enough to understand if you'd need it or not, and then use the man page as the main reference.